File: LeetCode/Hao/12IntegerToRoman.java
public class Solution {
    String[] ones = {
        "", "I","II","III","IV","V","VI","VII","VIII","IX",
    };
    String[] tens = {
        "", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"  
    };
    String[] hundreds = {
        "", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"
    };
    String[] thouands = {
        "", "M", "MM", "MMM", "MMMM"
    };
    public String intToRoman(int num) {
        String result = "";
        
        
        result += thouands[num/1000] + hundreds[num / 100 % 10] + tens[num / 10 % 10] + ones[num % 10];
        
        return result;
    }
}

File: LeetCode/Peng/12IntegerToRoman.cpp
class Solution {
public:
    string intToRoman(int num) {
        unordered_map<int, string> dict = {{1000, "M"}, {900, "CM"}, {500, "D"}, {400, "CD"}, {100, "C"}, {90, "XC"}, {50, "L"}, 
                                           {40, "XL"}, {10, "X"}, {9, "IX"}, {5, "V"}, {4, "IV"}, {1, "I"}};
        int candidates[13] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        string result = "";
        for (int i = 0; i < 13; i++) {
            int scale = candidates[i];
            while (num >= scale) {
                result += dict.at(scale);
                num -= scale;
            }
        }
        return result;
    }
};